pkg/client/queue/get_all.go

```diff
@@ -0,0 +1,87 @@
+package queue
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/armadaproject/armada/pkg/api"
+	"github.com/armadaproject/armada/pkg/client"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type GetAllAPI func() ([]*api.Queue, error)
+
+func GetAll(getConnectionDetails client.ConnectionDetails) GetAllAPI {
+	return func() ([]*api.Queue, error) {
+		connectionDetails, err := getConnectionDetails()
+		if err != nil {
+			return nil, fmt.Errorf("failed to obtain api connection details: %s", err)
+		}
+		conn, err := client.CreateApiConnection(connectionDetails)
+		if err != nil {
+			return nil, fmt.Errorf("failed to connect to api because %s", err)
+		}
+		defer conn.Close()
+
+		client := api.NewSubmitClient(conn)
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		allQueues := make([]*api.Queue, 0)
+		queueStream, err := client.GetQueues(ctx, &api.StreamingQueueGetRequest{})
+		if err != nil {
+			log.Error(err)
+			return nil, err
+		}
+
+		for {
+
+			msg, e := queueStream.Recv()
+			if e != nil {
+				if err, ok := status.FromError(e); ok {
+					switch err.Code() {
+					case codes.NotFound:
+						log.Error(err.Message())
+						return nil, e
+					case codes.PermissionDenied:
+						log.Error(err.Message())
+						return nil, e
+					}
+				}
+				if e == io.EOF {
+					return nil, e
+				}
+				if !isTransportClosingError(e) {
+					log.Error(e)
+				}
+				time.Sleep(5 * time.Second)
```

Comment:
I'm not sure we should do this.  When we watch events this is so that we can reconnect if there is a transient error, which is probably reasonable as watching events can take ages.  Getting queues should be quick, however so I'd say we should just fail immediately if there is an issue, rather than having the client hang for 5 seconds.